src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -623,8 +624,10 @@ internal Boolean LoadLocalHeaderExtraFieldAndCompressedBytesIfNeeded()
                 _compressedBytes = new Byte[(_compressedSize / MaxSingleBufferSize) + 1][];
                 for (int i = 0; i < _compressedBytes.Length - 1; i++)
                 {
-                    _compressedBytes[i] = new Byte[MaxSingleBufferSize];
+                    _compressedBytes[i] = ArrayPool<byte>.Shared.Rent(MaxSingleBufferSize);
```

Comment:
Adding a finalizer itself has cost, even if the instance is disposed and finalization suppressed.  But the type doesn't even implement IDisposable, so no code today is going to be disposing of it, which means we'd likely end up leaving a lot of instances to be cleaned up during finalization, which is relatively expensive.

Might be better to just not use the pool here, if we're concerned about enough of the buffers not being returned.